CI (sometimes) fails on macOS

Sometimes it's this

```
 error[E0463]: can't find crate for `pear_codegen` which `pear` depends on
  --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_http-0.4.5/src/lib.rs:15:14
   |
15 | #[macro_use] extern crate pear;
   |              ^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `rocket_http`
```

While other times it's a similar thing but it can't find serde_derive or serde or whatever. I have no idea why and it's even more strange that it only occurs on macOS as it doesn't seem to have to do with anything platform specific.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,6 +18,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Set up Rust nightly
+      id: toolchain
       uses: actions-rs/toolchain@v1
       with:
         profile: minimal
@@ -26,12 +27,13 @@ jobs:
     - name: Cache cargo build
       uses: actions/cache@v1
       env:
-        cache-name: cargo-build-target-2
+        cache-name: cargo-build-target
       with:
         path: target
-        key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('**/Cargo.lock') }}
+        # key needs to contain rustc_hash due to https://github.com/ActivityWatch/aw-server-rust/issues/180
+        key: ${{ runner.os }}-${{ env.cache-name }}-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
         restore-keys: |
-          ${{ runner.os }}-${{ env.cache-name }}-
+          ${{ runner.os }}-${{ env.cache-name }}-${{ steps.toolchain.outputs.rustc_hash }}-
     - name: Build
       run: cargo build --workspace --verbose
     - name: Run tests

